modules/system/classes/CompilableAssets.php

```diff
@@ -0,0 +1,307 @@
+                }
+            }
+
+            // Search modules for compilable packages to autoregister
+            foreach ($enabledModules as $module) {
+                $module = strtolower($module);
+                $path = base_path('modules' . DIRECTORY_SEPARATOR . $module) . DIRECTORY_SEPARATOR . $config['configFile'];
+                if (File::exists($path)) {
+                    $packagePaths[$type]["module-$module"] = $path;
+                }
+            }
+
+            // Search plugins for compilable packages to autoregister
+            $plugins = PluginManager::instance()->getPlugins();
+            foreach ($plugins as $plugin) {
+                $path = $plugin->getPluginPath() . '/' . $config['configFile'];
+                if (File::exists($path)) {
+                    $packagePaths[$type][$plugin->getPluginIdentifier()] = $path;
+                }
+            }
+
+            // Search themes for compilable packages to autoregister
+            if (in_array('Cms', $enabledModules)) {
+                $themes = Theme::all();
+                foreach ($themes as $theme) {
+                    $path = $theme->getPath() . '/' . $config['configFile'];
+                    if (File::exists($path)) {
+                        $packagePaths[$type]["theme-" . $theme->getId()] = $path;
+                    }
+                }
+            }
+        }
+
+        // Register the autodiscovered compilable packages
+        foreach ($packagePaths as $type => $packages) {
+            foreach ($packages as $package => $path) {
+                try {
+                    $this->registerPackage($package, $path, $type);
+                } catch (SystemException $e) {
+                    // Either the package name or the config file path have already been registered, skip.
+                    continue;
+                }
+            }
+        }
+    }
+
+    /**
+     * Register a compilable config, this allows the
+     * @param string $name
+     * @param array $config
+     * @return void
+     */
+    public function registerCompilable(string $name, array $config): void
+    {
+        $this->compilableConfigs[$name] = $config;
+    }
+
+    /**
+     * Registers a callback for processing.
+     */
+    public static function registerCallback(callable $callback): void
+    {
+        static::$callbacks[] = $callback;
+    }
+
+    /**
+     * Calls the deferred callbacks.
+     */
+    public function fireCallbacks(): void
+    {
+        // Call callbacks
+        foreach (static::$callbacks as $callback) {
+            $callback($this);
+        }
+    }
+
+    /**
+     * Returns the count of packages registered.
+     */
+    public function getPackageCount(): int
+    {
+        return array_sum(array_map(fn ($packages) => count($packages), ...$this->packages));
+    }
+
+    /**
+     * Returns all packages registered.
+     */
+    public function getPackages(bool $includeIgnored = false, string $type = 'mix'): array
```

Comment:
Kind of want to swap the arg order here, not sure if anything relies on it external to the core

Reply:
Shouldn't be anything besides the Mix/Vite commands using this, so should be safe to swap.

Reply:
I'm fine with swapping the order